pageserver/src/tenant/mgr.rs

```diff
@@ -251,6 +251,8 @@ pub async fn shutdown_all_tenants() {
                 tenants_clone
             }
             TenantsMap::ShuttingDown(_) => {
+                // TODO: it is possible that detach and shutdown happen at the same time. as a
```

Comment:
Lets have an issue for that if we dont have one already

Reply:
After a bit of pondering, thinking that the `shutdown_tasks(None, None, None);` will wait for that, but I'll create an issue.

Reply:
Created #4429